packages/headless/src/controllers/insight/smart-snippet/headless-insight-smart-snippet-interactive-inline-links.ts

```diff
@@ -0,0 +1,129 @@
+import {InsightEngine} from '../../../app/insight-engine/insight-engine';
+import {questionAnswering, search} from '../../../app/reducers';
+import {
+  logOpenSmartSnippetInlineLink,
+  logOpenSmartSnippetSuggestionInlineLink,
+} from '../../../features/question-answering/question-answering-insight-analytics-actions';
+import {
+  QuestionAnsweringSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {getObjectHash} from '../../../utils/utils';
+import {
+  buildInteractiveResultCore,
+  InteractiveResultCore,
+} from '../../core/interactive-result/headless-core-interactive-result';
+import {InlineLink} from '../../smart-snippet/headless-smart-snippet-interactive-inline-links';
+
+export interface SmartSnippetInteractiveInlineLinksOptions {
+  selectionDelay?: number;
+}
+
+export interface SmartSnippetInteractiveInlineLinksProps {
+  options?: SmartSnippetInteractiveInlineLinksOptions;
+}
+
+export interface SmartSnippetInteractiveInlineLinks {
+  selectInlineLink(link: InlineLink, questionAnswerId?: string): void;
+  beginDelayedSelectInlineLink(
+    link: InlineLink,
+    questionAnswerId?: string
+  ): void;
+  cancelPendingSelectInlineLink(
+    link: InlineLink,
+    questionAnswerId?: string
+  ): void;
+}
+
```

Comment:
should we add a comment here with the params and the return value like it is done for the smart-snippet controller?

Reply:
Those are not exposed outside of the internal code of headless